language/stdlib/modules/libra_system.mvir

```diff
@@ -229,6 +232,36 @@ module LibraSystem {
         return;
     }
 
+    public remove_validator(account_address: address) acquires ValidatorSet {
+        let validator_set_ref: &mut Self.ValidatorSet;
+
+        // Only the Association can remove validators
+        assert(get_txn_sender() == 0xA550C18, 1);
+
+        validator_set_ref = borrow_global_mut<ValidatorSet>(0x1D8);
+        // Ensure that this address is already a validator
+        assert(
+            Self.is_validator_(&account_address, &copy(validator_set_ref).validators),
+            21
+        );
```

Comment:
If I understand correctly, this is making sure that the validator being removed is already in the validator set, _but_ we can't remove a validator once it's been added to the `additions` list, but not yet become a part of the validator set.  Do we want to permit removing a validator that has been added to the `additions`  list, but not yet added as a validator?

Reply:
> If I understand correctly, this is making sure that the validator being removed is already in the validator set, but we can't remove a validator once it's been added to the additions list, but not yet become a part of the validator set

That is correct.

> . Do we want to permit removing a validator that has been added to the additions list, but not yet added as a validator?

My thought was that the simplest semantics is one where additions and removals refer to the existing validator set, not the pending validator set. Otherwise, you get into a lot of questions that don't have obvious answers (i.e., "should removals override additions")?

Reply:
Agreed that this provides the simplest semantics, and I think this is perfectly fine -- just wanted to make sure that this was the intended semantics.

Reply:
Because `removals` and `additions` don't interact, this is covered by the existing tests for `add` and `remove`. Specifically:
- Trying to add a validator that already exists in the validator set *or* is already in the pending additions (see add_validator.mvir)
- Trying to remove a validator that's not in the validator set *or* is already in the pending removals (see remove_validator.mvir)

Remove + add (or add + remove) will hit one of those cases depending on whether the address in question is already in the validator set or not.